### PR TITLE
CodeQL: Fix Go 1.21 build fix and update branch list

### PR DESCRIPTION
**Pull request by jentfoo**
Fix CodeQL Build and update branch list to include the v14 branch.

This PR fixes https://github.com/gravitational/SecOps/issues/414

Test build and analysis can be seen in this run: https://github.com/gravitational/teleport/actions/runs/6188331568/job/16800106312?pr=31892

### CODE REVIEW

File: .github/workflows/codeql.yml
Comment (codingllama) [inline .github/workflows/codeql.yml:20]: Should we keep branch/v11 for now?

**Review by codingllama — APPROVED** [review]

File: .github/workflows/codeql.yml
Comment (wadells) [reply .github/workflows/codeql.yml:20]: For CodeQL I think it is safe to drop v11.

We're getting little negative signal out of these the recurring checks AFAIK.  The hassle of remembering to remove v11 is a greater cost to the org than the chance we miss something important on v11 in the next week.

**Review by wadells — APPROVED** [review]

File: .github/workflows/codeql.yml
Comment (jentfoo) [reply .github/workflows/codeql.yml:20]: @wadells is echo'ing my thoughts as well.  It's a small time window and any changes in v11 should go through other branches first anyways.  So the window of risk is small and since this is a workflow update this PR needs significant approval: https://github.com/gravitational/teleport/blob/master/.github/CODEOWNERS#L3

**Review by rosstimothy — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
